### ISSUE TITLE
HTTP2FaxBridge

### ISSUE THREAD

**Issue by FaxHt** [sagiegurari/fax4j#5]
Hi Sagie,

I am trying to send fax using HTTP2FaxBridge using Web2FaxHttpServlet But I am getting INVALID_DAT error. Below is my code. Please help me if I am missing anything. Thanks.

private static void sendFaxHttpReg() throws ClientProtocolException, IOException {
		String url =
		         "http://localhost:8080/Web2Fax/Web2Fax";
		DefaultHttpClient httpclient = new DefaultHttpClient();
		HttpPost httppost = new HttpPost(url);

		FileBody bin = new FileBody(new File("/faxtest.txt"));
		StringBody targetaddress = new StringBody("1234");
		StringBody filename = new StringBody("faxtest");
		
		MultipartEntity reqEntity = new MultipartEntity();
		reqEntity.addPart("targetaddress", targetaddress);
		reqEntity.addPart("file", bin);
		reqEntity.addPart("filename", filename);
		reqEntity.addPart("targetname", filename);
		
		
		httppost.setEntity(reqEntity);

		HttpResponse response = httpclient.execute(httppost);
		HttpEntity resEntity = response.getEntity();
	     
	}

**Comment by sagiegurari** [comment]
I published this code 8 years ago in sourceforge. can barely remember whats going on with fax4j :)
But i'm guessing the error is on the http server side that holds the servlet.
if so, i need the fax4j log output. The javadocs got a section explaining how to enable the fax4j debug output. That might help.

**Comment by FaxHt** [comment]
Below is the log output.

org.fax4j.FaxException: Error while invoking fax4j.exe, error: Unable to send fax - ERROR_INVALID_DATA
Native Output:
[fax4j-native][DEBUG] Debug mode enabled.
[fax4j-native][DEBUG] Checking argument:
[fax4j-native][DEBUG] -action
[fax4j-native][DEBUG] Argument value:
[fax4j-native][DEBUG] submit
[fax4j-native][DEBUG] Checking argument:
[fax4j-native][DEBUG] -target_address
[fax4j-native][DEBUG] Argument value:
[fax4j-native][DEBUG] 8273512
[fax4j-native][DEBUG] Checking argument:
[fax4j-native][DEBUG] -target_name
[fax4j-native][DEBUG] Argument value:
[fax4j-native][DEBUG] KP
[fax4j-native][DEBUG] Checking argument:
[fax4j-native][DEBUG] -file
[fax4j-native][DEBUG] Argument value:
[fax4j-native][DEBUG] C:\AppData\Local\Temp\fax_5373501888128907003KP
[fax4j-native][DEBUG] Read input:
[fax4j-native][DEBUG] Action:
[fax4j-native][DEBUG] submit
[fax4j-native][DEBUG] Server:
[fax4j-native][DEBUG] (null)
[fax4j-native][DEBUG] Fax Job ID:
[fax4j-native][DEBUG] (null)
[fax4j-native][DEBUG] Target Address:
[fax4j-native][DEBUG] 8273512
[fax4j-native][DEBUG] Target Name:
[fax4j-native][DEBUG] KP
[fax4j-native][DEBUG] Sender Name:
[fax4j-native][DEBUG] (null)
[fax4j-native][DEBUG] File:
[fax4j-native][DEBUG] C:\AppData\Local\Temp\fax_5373501888128907003KP
[fax4j-native][DEBUG] Document Name:
[fax4j-native][DEBUG] (null)
[fax4j-native][DEBUG] Handling specific fax action.
[fax4j-native][DEBUG] Invoking submitFaxJobNative
[fax4j-native][DEBUG] Connecting to fax server.
[fax4j-native][DEBUG] Initializing fax job parameters.
[fax4j-native][DEBUG] Releasing cover page info.
[fax4j-native][DEBUG] Setting up fax job parameters.
[fax4j-native][DEBUG] Target Address:
[fax4j-native][DEBUG] 8273512
[fax4j-native][DEBUG] Sending fax with file:
[fax4j-native][DEBUG] C:\AppData\Local\Temp\fax_5373501888128907003KP
[fax4j-native][DEBUG] Error Code: 13 Message: Unable to send fax - ERROR_INVALID_DATA 
[fax4j-native][DEBUG] Releasing fax job parameters.
[fax4j-native][DEBUG] Closing connection to fax server.
fax.operation.done=false
fax.error=Unable to send fax - ERROR_INVALID_DATA

	at org.fax4j.spi.windows.WindowsProcessOutputValidator.validateProcessOutput(WindowsProcessOutputValidator.java:87)
	at org.fax4j.spi.process.ProcessFaxClientSpi.validateProcessOutput(ProcessFaxClientSpi.java:566)
	at org.fax4j.spi.process.ProcessFaxClientSpi.executeProcess(ProcessFaxClientSpi.java:492)
	at org.fax4j.spi.process.ProcessFaxClientSpi.submitFaxJobImpl(ProcessFaxClientSpi.java:351)
	at org.fax4j.spi.AbstractFaxClientSpi.submitFaxJob(AbstractFaxClientSpi.java:308)
	at org.fax4j.spi.adapter.AbstractAdapterFaxClientSpi.submitFaxJobImpl(AbstractAdapterFaxClientSpi.java:94)
	at org.fax4j.spi.AbstractFaxClientSpi.submitFaxJob(AbstractFaxClientSpi.java:308)
	at org.fax4j.spi.adapter.AbstractAdapterFaxClientSpi.submitFaxJobImpl(AbstractAdapterFaxClientSpi.java:94)
	at org.fax4j.spi.AbstractFaxClientSpi.submitFaxJob(AbstractFaxClientSpi.java:308)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
	at java.lang.reflect.Method.invoke(Method.java:597)
	at org.fax4j.spi.AbstractFaxClientSpiProxy.invoke(AbstractFaxClientSpiProxy.java:203)
	at $Proxy242.submitFaxJob(Unknown Source)
	at org.fax4j.FaxClient.submitFaxJob(FaxClient.java:214)
	at org.fax4j.bridge.AbstractFaxBridge.submitFaxJob(AbstractFaxBridge.java:265)
	at org.fax4j.bridge.AbstractContextFaxBridge.submitFaxJob(AbstractContextFaxBridge.java:91)
	at org.fax4j.bridge.FaxBridgeFlowHelper.submitFaxJob(FaxBridgeFlowHelper.java:83)
	at org.fax4j.x2fax.http.servlet.AbstractWeb2FaxHttpServlet.doPost(AbstractWeb2FaxHttpServlet.java:297)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:637)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:717)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:290)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
	at org.jboss.web.tomcat.filters.ReplyHeaderFilter.doFilter(ReplyHeaderFilter.java:96)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:235)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:235)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:191)
	at org.jboss.web.tomcat.security.SecurityAssociationValve.invoke(SecurityAssociationValve.java:190)
	at org.jboss.web.tomcat.security.JaccContextValve.invoke(JaccContextValve.java:92)
	at org.jboss.web.tomcat.security.SecurityContextEstablishmentValve.process(SecurityContextEstablishmentValve.java:126)
	at org.jboss.web.tomcat.security.SecurityContextEstablishmentValve.invoke(SecurityContextEstablishmentValve.java:70)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:127)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:102)
	at org.jboss.web.tomcat.service.jca.CachedConnectionValve.invoke(CachedConnectionValve.java:158)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:109)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:330)
	at org.apache.coyote.http11.Http11Processor.process(Http11Processor.java:829)
	at org.apache.coyote.http11.Http11Protocol$Http11ConnectionHandler.process(Http11Protocol.java:598)
	at org.apache.tomcat.util.net.JIoEndpoint$Worker.run(JIoEndpoint.java:447)
	at java.lang.Thread.run(Thread.java:662)

**Comment by sagiegurari** [comment]
can you please add extension to the file name? it helps windows discover if it is able to print it

**Comment by FaxHt** [comment]
Got a different exception after I added a extension to filename.

2017-11-02 14:47:59,339 ERROR [org.apache.catalina.core.ContainerBase.[jboss.web].[localhost].[/Web2Fax].[Web2FaxHttpServlet]] (http-127.0.0.1-8080-4) Servlet.service() for servlet Web2FaxHttpServlet threw exception
org.fax4j.FaxException: Unable to part multi part request.
	at org.fax4j.x2fax.http.servlet.AbstractWeb2FaxHttpServlet.convertMultiPartRequest(AbstractWeb2FaxHttpServlet.java:202)
	at org.fax4j.x2fax.http.servlet.AbstractWeb2FaxHttpServlet.convertRequest(AbstractWeb2FaxHttpServlet.java:230)
	at org.fax4j.x2fax.http.servlet.AbstractWeb2FaxHttpServlet.doPost(AbstractWeb2FaxHttpServlet.java:294)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:637)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:717)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:290)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
	at org.jboss.web.tomcat.filters.ReplyHeaderFilter.doFilter(ReplyHeaderFilter.java:96)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:235)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:235)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:191)
	at org.jboss.web.tomcat.security.SecurityAssociationValve.invoke(SecurityAssociationValve.java:190)
	at org.jboss.web.tomcat.security.JaccContextValve.invoke(JaccContextValve.java:92)
	at org.jboss.web.tomcat.security.SecurityContextEstablishmentValve.process(SecurityContextEstablishmentValve.java:126)
	at org.jboss.web.tomcat.security.SecurityContextEstablishmentValve.invoke(SecurityContextEstablishmentValve.java:70)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:127)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:102)
	at org.jboss.web.tomcat.service.jca.CachedConnectionValve.invoke(CachedConnectionValve.java:158)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:109)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:330)
	at org.apache.coyote.http11.Http11Processor.process(Http11Processor.java:829)
	at org.apache.coyote.http11.Http11Protocol$Http11ConnectionHandler.process(Http11Protocol.java:598)
	at org.apache.tomcat.util.net.JIoEndpoint$Worker.run(JIoEndpoint.java:447)
	at java.lang.Thread.run(Thread.java:662)
Caused by: org.apache.commons.fileupload.FileUploadBase$IOFileUploadException: Processing of multipart/form-data request failed. Stream ended unexpectedly
	at org.apache.commons.fileupload.FileUploadBase.parseRequest(FileUploadBase.java:351)
	at org.apache.commons.fileupload.servlet.ServletFileUpload.parseRequest(ServletFileUpload.java:115)
	at org.fax4j.x2fax.http.servlet.AbstractWeb2FaxHttpServlet.convertMultiPartRequest(AbstractWeb2FaxHttpServlet.java:176)
	... 24 more
Caused by: org.apache.commons.fileupload.MultipartStream$MalformedStreamException: Stream ended unexpectedly
	at org.apache.commons.fileupload.MultipartStream$ItemInputStream.makeAvailable(MultipartStream.java:1003)
	at org.apache.commons.fileupload.MultipartStream$ItemInputStream.read(MultipartStream.java:901)
	at java.io.InputStream.read(InputStream.java:82)
	at org.apache.commons.fileupload.util.Streams.copy(Streams.java:101)
	at org.apache.commons.fileupload.util.Streams.copy(Streams.java:70)
	at org.apache.commons.fileupload.FileUploadBase.parseRequest(FileUploadBase.java:347)
	... 26 more

**Comment by sagiegurari** [comment]
Caused by: org.apache.commons.fileupload.MultipartStream$MalformedStreamException: Stream ended unexpectedly

this is not fax4j related

**Comment by FaxHt** [comment]
Hi Sagie,

Thanks for immediate response. My requirement is to fax document from application (no fax machine on client side). So I am trying use Http2FaxBrige by using Web2FaxHttpServlet (web2fax.war component). Is it a right approach.

Thanks.

**Comment by sagiegurari** [comment]
generally speaking it was meant for that. but later I was thinking why would anyone expose that servlet as is. more likely people would have their own business logic and then invoke the FaxClient in their own code (servlet, bean, whatever).
It did work, just don't have java available on my machine these days, so can't really give an example.
maybe got working tests or something there to look at.

**Comment by sagiegurari** [comment]
another approach i was think when developing, is that the client would ALSO use fax4j.
fax4j has http client support, so use that for client and have fax4j on server machine to do the actual fax communication with the fax modem.
But really hard to give you a sample code.

**Comment by FaxHt** [comment]
I doubt if I can apply that approach in my requirement. My SLA is to send fax from Java code ( without using fax modem on sending side) to receipt Fax Machine. Any sample code to send httprequest to Web2Fax.war helps

**Comment by sagiegurari** [comment]
ya, but you can use fax4j on client using http spi to send the http request to the servlet

**Comment by FaxHt** [comment]
Thanks. Let me try that approach. It helps incase if you have any sample code.

**Comment by sagiegurari** [comment]
docs for fax client using http transport
in https://sagiegurari.github.io/fax4j/apidocs/overview-summary.html#overview_description
check 
org.fax4j.spi.http.HTTPFaxClientSpi
org.fax4j.spi.http.faxjob2request.converter.class.name - org.fax4j.spi.http.MultiPartFaxJob2HTTPRequestConverter

Test code that might help
https://github.com/sagiegurari/fax4j/blob/master/src/test/java/org/fax4j/spi/http/HTTPFaxClientSpiTest.java
https://github.com/sagiegurari/fax4j/blob/master/src/test/java/org/fax4j/spi/http/MultiPartFaxJob2HTTPRequestConverterTest.java

Can't do much more than that. Like I said, has been many years since I wrote this and since I worked with java.

**Comment by FaxHt** [comment]
No Problem. Thank you so much. I will try with it.

**Comment by FaxHt** [comment]
Hi Sagie,

Sorry for bothering you. I tried using spi to send request to servlet now I am not getting any exception or any log. It opens up the pdf in adobe reader and status returns as UNKNOWN. And I didn't get any fax. Do you have any idea why it happens. Thanks.

**Comment by FaxHt** [comment]
This is code I am trying...

package org.fax4j.spi.http;

import java.io.File;
import java.util.Properties;

import org.fax4j.FaxJob;
import org.fax4j.FaxJob.FaxJobPriority;
import org.fax4j.common.FaxActionType;
import org.fax4j.spi.FaxJobImpl;
import org.fax4j.spi.http.HTTPRequest.ContentPart;
import org.fax4j.test.TestUtil;
import org.fax4j.test.TestUtil.TestHTTPClient;


public class FaxClientSpiTest {
	    private HTTPFaxClientSpi faxClientSpi;
	    /**The configuration*/
	    private Properties configuration;

	    private MultiPartFaxJob2HTTPRequestConverter converter;

	public static void main(String[] args) throws Exception {
		System.out.println("test Spi");
		FaxClientSpiTest faxClientSpiTest=new FaxClientSpiTest();
		faxClientSpiTest.setUpMultiPart();
		faxClientSpiTest.createSubmitFaxJobHTTPRequestTest();
	}
	 
	   
	 
	    
	    public void setUpMultiPart() throws Exception
	    {
	        this.configuration=new Properties();
	        this.configuration.setProperty("org.fax4j.spi.http.client.class.name",TestHTTPClient.class.getName());
	        this.configuration.setProperty("org.fax4j.spi.http.faxjob2request.converter.class.name",MultiPartFaxJob2HTTPRequestConverter.class.getName());
	       // this.configuration.setProperty("org.fax4j.spi.http.response.handler.class.name",EmptyHTTPResponseHandler.class.getName());
	        this.configuration.setProperty("org.fax4j.spi.http.host.name","localhost");
	        this.configuration.setProperty("org.fax4j.spi.http.port","8080");
	        this.configuration.setProperty("org.fax4j.spi.http.ssl","false");
	        this.configuration.setProperty("org.fax4j.spi.http.submit.resource","/Web2Fax/Web2Fax");
	       // this.configuration.setProperty("org.fax4j.spi.http.url.parameters","param1=value1&param2=value2");

	        this.configuration.setProperty("org.fax4j.spi.http.multi.part.submit.file.content.parameter","file");
	        this.configuration.setProperty("org.fax4j.spi.http.multi.part.submit.file.name.parameter","filename");
	        this.configuration.setProperty("org.fax4j.spi.http.multi.part.submit.target.address.parameter","targetaddress");
	        this.configuration.setProperty("org.fax4j.spi.http.multi.part.submit.target.name.parameter","targetname");
	        this.configuration.setProperty("org.fax4j.spi.http.multi.part.submit.sender.name.parameter","sendername");

	        this.configuration.setProperty("org.fax4j.spi.http.multi.part.submit.sender.fax.number.parameter","senderfaxnumber");
	       // this.configuration.setProperty("org.fax4j.spi.http.multi.part.submit.sender.email.parameter","test_senderemail");
	       // this.configuration.setProperty("org.fax4j.spi.http.multi.part.suspend.fax.job.id.parameter","test_suspend_faxjobid");
	       // this.configuration.setProperty("org.fax4j.spi.http.multi.part.resume.fax.job.id.parameter","test_resume_faxjobid");
	       // this.configuration.setProperty("org.fax4j.spi.http.multi.part.cancel.fax.job.id.parameter","test_cancel_faxjobid");
	      //  this.configuration.setProperty("org.fax4j.spi.http.multi.part.get.status.fax.job.id.parameter","test_getstatus_faxjobid");
	       // this.configuration.setProperty("org.fax4j.spi.http.multi.part.parameter.key.KEY1","VALUE1");
	      //  this.configuration.setProperty("org.fax4j.spi.http.multi.part.parameter.key.KEY2","VALUE2");
	        
	        this.faxClientSpi=(HTTPFaxClientSpi)TestUtil.createFaxClientSpi(HTTPFaxClientSpi.class.getName(),this.configuration);
	        this.converter=(MultiPartFaxJob2HTTPRequestConverter)this.faxClientSpi.getFaxJob2HTTPRequestConverter();

	    }
	    public void getHTTPResponseHandlerTest() throws Exception
	    {
	    	String value=this.faxClientSpi.getHTTPURLParameters();
	        System.out.print(value);
	    }
	    public void createSubmitFaxJobHTTPRequestTest() throws Exception
	    {
	        File file=new File("C:/Users/Documents/KP.pdf");
	         FaxJob faxJob=new FaxJobImpl();
	        faxJob.setFile(file);
	        System.out.println(file.getPath());
	        faxJob.setID("test1");
	        faxJob.setFilePath(file.getPath());
	         faxJob.setPriority(FaxJobPriority.HIGH_PRIORITY);
	        faxJob.setTargetAddress("8888273512");
	        faxJob.setTargetName("/Web2Fax/Web2Fax");
	        faxJob.setSenderName("test");
	        faxJob.setSenderFaxNumber("827-3512");
	      //  HTTPRequest output=this.faxClientSpi.createSubmitFaxJobHTTPRequest(faxJob);
	        HTTPRequest output=this.converter.createHTTPRequest(this.faxClientSpi,FaxActionType.SUBMIT_FAX_JOB,faxJob);

	        this.faxClientSpi.submitHTTPRequest(faxJob, output, FaxActionType.SUBMIT_FAX_JOB);
	        ContentPart<?>[] parts=output.getContentAsParts();
	        for(int index=0;index<parts.length;index++)
	        {
	            ContentPart<?> part=parts[index];
	            System.out.println(part.getName() +"----"+part.getContent());
	            
	        }
	        System.out.println(this.faxClientSpi.getFaxJobStatus(faxJob));
		}

}

**Comment by sagiegurari** [comment]
i thought it was txt, not pdf.
windows fax is based on print capability and adobe reader is not great to say the least for that.
you should install foxit reader instead. it free and it works much better.

**Comment by FaxHt** [comment]
I tried with txt also it returns status  as UNKNOWN

**Comment by sagiegurari** [comment]
lets take a step back.
first run some test code and see if you can even fax something in that server machine.
no server, no http. just simple java code to send fax like in the example in the main readme.
just to make sure that the machine is setup good.

**Comment by FaxHt** [comment]
I ran the below sample script status is pending. But as I understand if I don't have a fax machine setup (sender side) then we should use X2Fax bridges right. And I am sending fax to eFax account number not to actual physical machine.

               FaxClient faxClient=FaxClientFactory.createFaxClient();
	        FaxJob faxJob=faxClient.createFaxJob();
                faxJob.setFile(new File("C:/Users/htirumuru/Documents/faxtest.txt"));
		faxJob.setPriority(FaxJobPriority.HIGH_PRIORITY);
		faxJob.setTargetAddress("661-8912");
		faxJob.setFilePath("C:/Users/Documents/faxtest.txt");
		faxClient.submitFaxJob(faxJob);

**Comment by sagiegurari** [comment]
Just want to clear what each component is:
1. fax client - this one actually does the sending of the fax using the provided fax job object.
you can send the fax using windows native api, http request (to some vendors who expose fax via rest), emails (for mail servers that support mail2fax conversion), run processes and so on...
2. To more easily expose the fax client using transport specific objects like httprequest for servlets, the fax bridge exists. basically the bridge is just a converter between transport objects like httprequest and fax job and than it uses the faxclient to send the fax. so bridges are just converters.
3. once you have bridges, the x2fax project exists to expose those bridges as servlets/cli and so on...

The project originated in sourceforge but once I moved to github I deleted the forum there (people kept asking me there and couldn't keep up so removed the forum there), but basically from my experience almost everyone uses the faxclient directly which makes sense, as  you probably have some API exposed via REST and you have some logic in that api like billing a customer for something and part of the flow is to send a fax, so using the fax client directly makes sense.
I'm less familiar of your use case, but

1. make sure that first faxclient works good on that server machine. if you sent and you are in pending status its because windows got and accepted the fax job, but it can't send it without fax modem so its stuck in its queue, hence pending state.
2. once that is done, you can expose that code however you like. servlet is one way (probably best way). you don't have to use the bridge or anything like that, just write your code to get the data and use the faxclient to submit the data. you can look at my code how I do it and write your own to make things work best for you, unless that servlet really does nothing except sending fax in which case, ya use x2fax.

**Comment by andreasinspire** [comment]
Sagie hello
Am using FAX4J to send faxes from a USB based fax modem on a windows 10 machine. When I run the code from eclipse it works fine. Faxes are sent. When I deploy on TOMCAT I get the following

org.fax4j.FaxException: Unable to send fax.
	at org.fax4j.spi.windows.WindowsJNIFaxClientSpi.submitFaxJobNative(Native Method)
	at org.fax4j.spi.windows.WindowsJNIFaxClientSpi.winSubmitFaxJob(WindowsJNIFaxClientSpi.java:274)
	at org.fax4j.spi.windows.WindowsJNIFaxClientSpi.submitFaxJobImpl(WindowsJNIFaxClientSpi.java:164)
	at org.fax4j.spi.AbstractFaxClientSpi.submitFaxJob(AbstractFaxClientSpi.java:308)
	at org.fax4j.spi.adapter.AbstractAdapterFaxClientSpi.submitFaxJobImpl(AbstractAdapterFaxClientSpi.java:94)
	at org.fax4j.spi.AbstractFaxClientSpi.submitFaxJob(AbstractFaxClientSpi.java:308)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.fax4j.spi.AbstractFaxClientSpiProxy.invoke(AbstractFaxClientSpiProxy.java:203)
	at com.sun.proxy.$Proxy113.submitFaxJob(Unknown Source)
	at org.fax4j.FaxClient.submitFaxJob(FaxClient.java:214)
	at inspireObjectsCommonFaxService.NA_FaxService.sendFax(NA_FaxService.java:77)
	at queueManagerClasses.JMSConsumer.subscribe(JMSConsumer.java:181)
	at queueManagerClasses.JMSConsumer.run(JMSConsumer.java:221)
	at java.lang.Thread.run(Unknown Source)
Can you advice please

**Comment by sagiegurari** [comment]
1. please open a new issue. It makes simpler for me and for others.
2. please provide full logs (see Error Handling section in https://sagiegurari.github.io/fax4j/apidocs/overview-summary.html#overview_description)

**Comment by andreasinspire** [comment]
Sagie thanks a lot for the immediate response

New issue opened and sent
